src/lib.rs

```diff
@@ -112,6 +114,32 @@ where
     Err(CredentialRetrievalError::NoCredentialConfigured)
 }
 
+/// Retrieve a user's docker credential from a given reader.
+///
+/// Example:
+/// ```no_run
+/// use std::{fs::File, io::BufReader};
+/// use docker_credential::DockerCredential;
+///
+/// let file = File::open("config.json").expect("Unable to open config file");
+///
+/// let reader = BufReader::new(file);
+///
+/// let credential = docker_credential::get_credential_from_reader(reader, "https://index.docker.io/v1/").expect("Unable to retrieve credential");
+///
+/// match credential {
+///   DockerCredential::IdentityToken(token) => println!("Identity token: {}", token),
+///   DockerCredential::UsernamePassword(user_name, password) => println!("Username: {}, Password: {}", user_name, password),
+/// };
+/// ```
+pub fn get_credential_from_reader(
```

Comment:
I am content with just the podman function being exposed. Initially I exposed the reader function to allow for arbitrary json lookup logic (to be used with Podman specifically), but then I decided to cover this functionality in this PR too. There are no other major container engines as of now, so the only major benefit would be to support docker configurations where the config name is not literally `config.json` (because the file name is not configurable, only the directory). It can hovewer be worked around using the new podman function, because the file path can be fully customized through the `REGISTRY_AUTH_FILE` env var there.

Reply:
Ok, lets keep it in to avoid a slightly hacky workaround for the case you mention